### DISCUSSION TITLE
NPM-PLUS issues with it going to sleep possibly

### DISCUSSION THREAD

**Discussion by Sabering1** [ZoeyVid/NPMplus#901] [Q&A] [answered]
duplicate of
https://github.com/nextcloud/all-in-one/issues/4877

**Comment by Zoey2936** [comment]
please also awnser the questions I asked. Thanks

**Comment by Sabering1** [comment]
I use nextcloud docker and community container called NPMplus, when everything is set up it works just fine, but after a while if its inactive i belive, it will bug out and the next time i try to load my nextcloud instance it will report a TLS error (not sure will attach a screenshot when it happens again), this will not happen with any other subdomains/services i have set up, which leads me to belive that maybe nextcloud stops working somehow? or that npm will go to sleep and stop listening on that port? i dont know i will provide the logs when it happens again.

I am running on a home server with static ip and all ports opened correctly and i can confirm that they are not blocked by my isp

**Reply by Sabering1** [reply]
```
[WARN  tini (7)] Tini is not running as PID 1 and isn't registered as a child subreaper.
Zombie processes will not be re-parented to Tini, so zombie reaping won't work.
To fix the problem, use the -s option or set the environment variable TINI_SUBREAPER to register Tini as a child subreaper, or run Tini as PID 1.
'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/README.md' -> '/data/etc/modsecurity/crs-plugins/README.md'
'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/empty-after.conf' -> '/data/etc/modsecurity/crs-plugins/empty-after.conf'
'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/empty-before.conf' -> '/data/etc/modsecurity/crs-plugins/empty-before.conf'
'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/empty-config.conf' -> '/data/etc/modsecurity/crs-plugins/empty-config.conf'
no DEFAULT_CERT_ID set, using dummycerts.
Working on file: /data/nginx/custom/events.conf
Working on file: /data/nginx/custom/http.conf
Working on file: /data/nginx/custom/http_top.conf
Working on file: /data/nginx/custom/root.conf
Working on file: /data/nginx/custom/server_dead.conf
Working on file: /data/nginx/custom/server_proxy.conf
Working on file: /data/nginx/custom/server_redirect.conf
Working on file: /data/nginx/custom/server_stream.conf
Working on file: /data/nginx/custom/server_stream_tcp.conf
Working on file: /data/nginx/custom/server_stream_udp.conf
Working on file: /data/nginx/custom/stream.conf
Working on file: /data/nginx/custom/stream_top.conf
Working on file: /data/nginx/default.conf
Working on file: /data/nginx/ip_ranges.conf
Working on file: /data/nginx/proxy_host/1.conf
Working on file: /data/nginx/proxy_host/10.conf
Working on file: /data/nginx/proxy_host/2.conf
Working on file: /data/nginx/proxy_host/3.conf
Working on file: /data/nginx/proxy_host/9.conf
Success.

-------------------------------------
 _ _  ___  __ __       _
| \ || . \|  \  \ ___ | | _ _  ___
|   ||  _/|     || . \| || | |[_-[
|_\_||_|  |_|_|_||  _/|_| \__|/__/
                 |_|
-------------------------------------
Version:  2.11.2+22fa2f0
Date:     Sat Jun 22 18:24:52 CEST 2024
User:     root
PUID:     0
User ID:  0
PGID:     0
Group ID: 0
-------------------------------------

nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
```

**Reply by Sabering1** [reply]
nextcloud shows also this : Some headers are not set correctly on your instance - The X-XSS-Protection HTTP header does not contain 1; mode=block. This is a potential security or privacy risk, as it is recommended to adjust this setting accordingly.

**Reply by Zoey2936** [reply]
> nextcloud shows also this : Some headers are not set correctly on your instance - The X-XSS-Protection HTTP header does not contain 1; mode=block. This is a potential security or privacy risk, as it is recommended to adjust this setting accordingly.

https://github.com/ZoeyVid/NPMplus/discussions/501

**Reply by Zoey2936** [reply]
for the tls thing I can first try to help you when I know about which error we are talking abount

**Reply by Sabering1** [reply]
sure i will post it here and alert you when i catch it happen

**Reply by Sabering1** [reply]
Okay it happend again
![image](https://github.com/ZoeyVid/NPMplus/assets/39810634/124f4f9a-d98e-4ea3-adb4-48b831cf9258)

**Reply by Sabering1** [reply]
![image](https://github.com/ZoeyVid/NPMplus/assets/39810634/4c5a7fec-d55a-4013-94c0-62338f1274e5)

**Reply by Zoey2936** [reply]
are you using a local dns resolver with dns rewrite?

**Reply by Sabering1** [reply]
i am using adguard DNS HOME , with some blocklist but no dns rewrites have been enabled

**Reply by Zoey2936** [reply, marked as answer]
Ok, try the following:
1. run `docker exec -it nextcloud-aio-npmplus nano /data/tls/certbot/config.ini`
2. change `must-staple = true` to `must-staple = false`
3. create a NEW certificate for `cloud.sabering.xyz`
4. change the certificate you use for you `cloud.sabering.xyz` to the NEW cert you created in last step
5. delete the OLD cert you used for `cloud.sabering.xyz`
6. report back if the error occured again or not

**Reply by Sabering1** [reply]
done, had to do it from scratch but everything seems to work fine for now, i will wait a day or two to see if it happens again, and if everything works good ill mark it as correct answer, prob also update the nextcloud documentation aswell

**Reply by Zoey2936** [reply]
if you search the discussions here this is an issue which often happens, most time disabling must-staple fixes it, but often there is no reason why it fixes it/why must-staple caused it and on nearly all systems it works

**Reply by Zoey2936** [reply]
can you maybe run `traceroute cloud.sabering.xyz` on one of your devices which is in the same lan as your server?

**Reply by Sabering1** [reply]
sure

**Reply by Sabering1** [reply]
sabering@fedora:~$ traceroute cloud.sabering.xyz
traceroute to cloud.sabering.xyz (109.111.249.99), 30 hops max, 60 byte packets
 1  free-249-99.mediaworksit.net (109.111.249.99)  5.418 ms  5.362 ms  5.340 ms
thats it 1 hop

**Reply by Zoey2936** [reply]
so there is the "issue" you are connecting "indrectly via lan" to you cloud, which breaks must-staple

**Reply by Sabering1** [reply]
oh well that makes so much more sense now, i think this should be mentioned under the nextcloud/community-containers/npmplus ?
might be useful to somebody?

**Reply by Zoey2936** [reply]
maybe I should disable must-stable for all new npmplus instances

**Reply by Sabering1** [reply]
i dont know up to you, but i think it being mentioned in the documentation would be usefull regardless
